crates/keys/src/key_manager/local_key_manager.rs

```diff
@@ -31,41 +38,81 @@ impl KeyManager for LocalKeyManager {
         curve: Curve,
         key_alias: Option<String>,
     ) -> Result<String, KeyManagerError> {
-        let key_alias = self.key_store.generate_new(curve, key_alias)?;
+        let private_key = Arc::new(match curve {
+            Curve::Ed25519 => Ed25519::generate(),
+            Curve::Secp256k1 => Secp256k1::generate(),
+        }?);
+        let key_alias = match key_alias {
+            Some(key_alias) => key_alias,
+            None => private_key.compute_thumbprint()?,
+        };
```

Comment:
Alternative, I don't care one way or the other 

```suggestion
        let key_alias = key_alias.ok_or_else(|| private_key.compute_thumbprint())?;
```

Reply:
ty! I prefer the `match` statement here